docs/data-guidelines/features.md

```diff
@@ -0,0 +1,55 @@
+
+A possible structure for a Web API is shown below:
+
+```
+category                // category name (always `api` for API interfaces)
+  interface               // top level feature
+    properties              // subfeature, add if supported
+      property values         // parameter subfeature, add all of them if not all values were supported since property was introduced
+    methods                // subfeature, add if supported
+      parameters              // parameter subfeature, add all of them if not all parameters were supported since method was introduced
+        parameter options       // parameter subfeature, add all of them if not all parameter options were supported since property was introduced
+        parameter type          // parameter subfeature, add all of them if not all parameter types were supported since property was introduced
+    support in workers     // behavioral subfeature, add if compatibility is different (might also just be on specific methods)
+```
+
+The interface must include subfeatures for _all_ its properties and methods that have an implementation in at least one of the supported browsers, whether or not they were added at the same time as the parent interface.
+
+All other code and behavioral aspects of the interface are given nested nodes **only** if their compatibility is _different_ to that of their parent feature. This applies to method arguments, method argument options, and behavior changes like the addition of support for workers, or the type of return value.
+
+> [!TIP]
+> A good rule of thumb is that if a behavioral subfeature has the same compatibility information as its parent, then the subfeature is not needed.
+> Another good rule of thumb is that if the signature of a method changes to introduce new parameters, then all parameters should be documented.
+
+Here is an example scenario:
+
+- An interface is defined in a specification with a method that has two parameters.
+- A supported browser implements the method and both the parameters:
+  - A feature node is created for the method (by default)
+  - Subfeature nodes are not created for the method parameters because they were created in the same version as the methods; there is no compatibility change.
+- Another browser implements the methods, but only one of the parameters.
+  - A subfeature node is added to the method parameter that was supported by _both_ browsers.
+  - Another subfeature node is added to the method parameter that was _not_ supported by the second browser.
+- Yet another browser implements the method with a new parameter.
+  - A method subfeature is added for the new parameter because it was added in a different version than the parent method.
+- The browser changes to require that the interface now can be used in workers:
+
+  - A "supported in workers" subfeature is added to the interface feature because this is a change in compatibility.
```

Comment:
Here's how I would restructure the example for better readability:

```suggestion
Given: A specification defines an interface with a method that has two parameters.

Assume: One supported browser implements the method and both the parameters.

Then:
- We create a feature node for the method (by default)
- We don't create subfeature nodes for the method parameters, because they were created in the same version as the methods; there is no compatibility change.

Assume: Another browser implements the method, but only one of the parameters.

Then:
- We add one subfeature node for the parameter that is supported by _both_ browsers.
- We add another subfeature node for the parameter that is only supported by the first, but _not_ by the second browser.

Assume: Yet another browser implements the method with a new parameter.

Then:
  - We add yet another subfeature node for the new parameter, because it was added in a different version than the parent method.
- 

Assume: One of the browsers starts supporting the interface in workers:

Then:
  - We add a `worker_support` subfeature (`"Available in workers"`) to the interface (see [Web Workers](./api.md#web-workers-worker_support)).
```

Reply:
@caugner I have no problem with this alternative approach, but I think I'd like the change the rule, not that I've been around a while.

The "rule" for API properties and methods is that you always add entries for all that are supported.

The current rule for method arguments (as I understand it and stated there) is that you only document the incompatibility. So the assumption is that if something is not documented then it is compatible.

I'd like to change the rule to "As soon as there is an incompatibility you should document all parameters including those that were in the original version.

The reason is that as it is now there is no way to spot whether something is compatible or has just been omitted.

Reply:
Upshot, if we know the rules I'm open to any presentation at all.

Reply:
If this PR explains the rule different than the current status quo, then let's document the status quo first, and then propose the change afterwards. That way we separate concerns and make it easier to move forward.

Anyways, I'll put this on the agenda for the next BCD call.

Reply:
Well, I wrote it to "test" the rules. So right now it reflects what I think the rules were, but not necessarily what they are.